### DISCUSSION TITLE
"Temporarily switch between first/current" could use second when it's first.

### DISCUSSION THREAD

**Discussion by HubKing** [fcitx/fcitx5#1038] [Ideas]
When the current input method is the first one, pressing the shortcut for "Temporarily switch between first/current" does nothing. Instead, I wish it switched to the second, or the last used non-first IME (for example, if the last used one was the third, switch to the third).

**Comment by wengxt** [comment]
Yeah, it was by design doing that :) ... I know the wording is probably confusing. Feel free to suggest other way of describing it.

The motivation is to simulate the temporarily English mode in Chinese Pinyin that is very common on other platforms. that is:
Normal case: under keyboard-... engine, and shift does nothing.
After you switch to an actual IM e.g. Pinyin, shift can be used to switch back to the keyboard-... (usually the first one).

The point is to not occupy the Shift if you haven't switch back to keyboard- with Shift.
